src/credentials/types.ts

```diff
@@ -0,0 +1,35 @@
+/*!
```

Comment:
Calling this file "types" is somewhat misleading but I can't really think of anything else.

Reply:
One thing I was thinking is that this would instead be named 'sharedCredentials.ts`, and the existing `sharedCredentials.ts` which holds the parsing related code of credentials could go to a new file like `sharedCredentialsParser.ts`

Reply:
I've changed my mind. `types.ts` is appropriate because `SharedCredentialsKeys` is being used as a enum. The typescript compiler follows a similar pattern.